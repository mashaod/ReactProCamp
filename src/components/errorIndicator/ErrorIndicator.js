import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import { Error } from '@material-ui/icons';

function styles(theme) {
    return {
        root: {
            position: "absolute",
            top: "50%",
            left: "50%",
            margin: "-50px 0px 0px -50px"
        }
    }
};

const ErrorIndicator = ({ message=null, classes }) => {
    const drowErrorIcon = () => (<Error className={classes.icon} color="error" fontSize="large" />)
    const drowErrorMessage = () => (<h3>{message}</h3>);

    return (
        <div className={classes.root} >
            { message ? drowErrorMessage() : drowErrorIcon() }
        </div>
    )
};

export default withStyles(styles)(ErrorIndicator);