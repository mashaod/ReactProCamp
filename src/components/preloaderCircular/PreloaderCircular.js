import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

function styles(theme) {
    return {
        root: {
            position: "absolute",
            top: "50%",
            left: "50%",
            transform: "translate(-50%,-50%)"
        }
    }
};

function PreloaderCircular({ classes }) {
    return (
        <div className={classes.root} >
            <CircularProgress
                variant="indeterminate"
                size={48}
            />
        </div>
    )
};

export default withStyles(styles)(PreloaderCircular);