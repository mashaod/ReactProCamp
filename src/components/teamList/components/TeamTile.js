
import React from 'react';
import { Link } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';

import Box from '@material-ui/core/Box';

const useStyles = makeStyles(theme => ({
    paper: {
        '&:hover': {
            transform: "scale(1.01)",
            transition: "0.3s",
            boxShadow: "0px 1px 12px 0px rgba(0,0,0,0.4), 0px 1px 1px 0px rgba(0,0,0,0.25), 0px 8px 5px -1px rgba(0,0,0,0.12)"
        }
    },
    control: {
      maxWidth: "250px"
    },
  }));

const TeamsByRowGroups = (props) => {
    const { team } = props;

    const classes = useStyles();

    return (
        <Link to={`/teams/${team.team_id}`} style={{ textDecoration: 'none', color: 'inherit' }}>
            <Box className={`${classes.control}, ${classes.paper}`} m={2} p={2}>
                <h2>{ team.name }</h2>
                <img src={team.logo} alt="team logo"></img>
                <p><i>{ team.venue_city }</i></p>
            </Box>
        </Link>
    )
};

export default TeamsByRowGroups;