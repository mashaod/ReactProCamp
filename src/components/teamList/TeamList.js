import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators, compose } from 'redux';
import { connect } from 'react-redux';

import { withAppService } from '../../hoc';
import { fetchTeams } from '../../store/actions/teams.actions';

import ErrorIndicator from '../errorIndicator';
import PreloaderCircular from '../preloaderCircular';
import TeamTile from './components/TeamTile';
import Grid from '@material-ui/core/Grid';

class TeamList extends Component {

    componentDidMount() {
        this.props.fetchTeams();
    }

    render() {
        const { teams, loading, error } = this.props;

        if (loading) { return <PreloaderCircular />; }
        if (error) { return <ErrorIndicator />; }

        return (
            <Grid container item xs={8} justify="center">
                {
                    teams.map((team) => (
                        <Grid key={team.team_id} item xs={3} align="center">
                            <TeamTile team={team}/>
                        </Grid>
                    ))
                }
            </Grid>
        )
    }
}

const mapStateToProps = ({ teamList: { teams, loading, error }}) => {
    return { teams, loading, error };
};

const mapDispatchToProps = (dispatch, { appService }) => {
    return bindActionCreators({
        fetchTeams: fetchTeams(appService),
    }, dispatch);
};

TeamList.propTypes = {
    teams: PropTypes.array.isRequired,
    loading: PropTypes.bool.isRequired
};

export default compose(
    withAppService(),
    connect(mapStateToProps, mapDispatchToProps)
)(TeamList);
