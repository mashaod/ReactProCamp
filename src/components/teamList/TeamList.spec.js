import React from 'react';
import { shallow } from 'enzyme';
import { TeamList } from './TeamList';

import ErrorIndicator from '../errorIndicator';
import PreloaderCircular from '../preloaderCircular';
import TeamTile from './components/TeamTile';

describe('TeamList container', () => {
    const props = {
        teams: [],
        loading: false,
        error: null,
        key: 1,
        onFetchTeams: () => {}
    }

    describe('TeamList container initial', () => {
        const mockFetchTeams = jest.fn()
        const nextProps = {
            ...props,
            onFetchTeams: mockFetchTeams
        }

        const teamListContainer = shallow(<TeamList {...nextProps} />);

        it('renders properly', () => {
            expect(teamListContainer).toMatchSnapshot();
        })

    })

    describe('TeamList container isLoading', () => {
        const nextProps = {
            ...props,
            loading: true
        }

        const teamListContainer = shallow(<TeamList {...nextProps} />);

        it('renders properly', () => {
            expect(teamListContainer).toMatchSnapshot();
        })

        it('renders preloader circular', () => {
            expect(teamListContainer.find(PreloaderCircular)).toHaveLength(1);
        })
    })

    describe('TeamList container render <TeamTile />', () => {
        const nextProps = {
            ...props,
            teams: [1]
        }

        const teamListContainer = shallow(<TeamList {...nextProps} />);

        it('renders properly', () => {
            expect(teamListContainer).toMatchSnapshot();
        })

        it('renders <TeamTile /> template', () => {
            expect(teamListContainer.find(TeamTile)).toHaveLength(1);
        })
    })

    describe('TeamList container with error', () => {
        const nextProps = {
            ...props,
            error: true
        }

        const teamListContainer = shallow(<TeamList {...nextProps} />);

        it('renders properly', () => {
            expect(teamListContainer).toMatchSnapshot();
        })

        it('renders error indicator', () => {
            expect(teamListContainer.find(ErrorIndicator)).toHaveLength(1);
        })
    })

})
