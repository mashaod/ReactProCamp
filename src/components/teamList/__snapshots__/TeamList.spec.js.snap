// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamList container TeamList container initial renders properly 1`] = `ShallowWrapper {}`;

exports[`TeamList container TeamList container isLoading renders properly 1`] = `ShallowWrapper {}`;

exports[`TeamList container TeamList container render <TeamTile /> renders properly 1`] = `ShallowWrapper {}`;

exports[`TeamList container TeamList container with error renders properly 1`] = `ShallowWrapper {}`;
